core/src/main/scala/org/bitcoins/core/util/Bech32.scala

```diff
@@ -75,7 +73,7 @@ sealed abstract class Bech32 {
     * [[https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#bech32 BIP173]]
     * rules
     */
-  def checkHrpValidity(hrp: String): Try[Bech32HumanReadablePart] = {
+  def checkHrpValidity(hrp: String): Try[String] = {
```

Comment:
I personally think this would be nice to have (at least as an overloaded option)
```
def checkHrpValidity(hrp: String, factories: Vector[StringFactory[Bech32HumanReadablePart]]]): Try[Bech32HumanReadablePart]
```
as this seems like an unnecessary loss of type seeing as we have a non-sealed abstract class that should be returned in most cases.
(note that this is going to require you make `StringFactory` co-variant (i.e. `[T]` -> `[+T]`) but it should already be that and I'm shocked that this hasn't previously caused issues).